dbcleaner-postgresql-simple/src/Database/DBCleaner/PostgreSQLSimple/Types.hs

```diff
@@ -1,12 +1,38 @@
+{-# LANGUAGE OverloadedStrings #-}
+
 module Database.DBCleaner.PostgreSQLSimple.Types where
 
-import qualified Database.PostgreSQL.Simple as PG
+import           Control.Monad                    (void)
+import           Data.Monoid                      (mconcat)
+import           Data.Text                        (Text)
+import           Database.PostgreSQL.Simple       (Connection, Only (..), Query,
+                                                   begin, execute, query_,
+                                                   rollback)
+import           Database.PostgreSQL.Simple.Types (Identifier (..))
+
+import           Database.DBCleaner.Types         (GenericConnection (..),
+                                                   Strategy (..))
+
+instance GenericConnection Connection where
+  before c Transaction = begin c
+  before c Truncation  = getTables c >>= mapM_ (truncateTable c)
+
+  after c Transaction = rollback c
+  after c Truncation  = before c Truncation
 
-import Database.DBCleaner.Types
+getTables :: Connection -> IO [Text]
+getTables c = map fromOnly `fmap` query_ c tablesQuery
 
-instance GenericConnection PG.Connection where
-  begin c Transaction = PG.begin c
-  begin c _           = undefined
+tablesQuery :: Query
```

Comment:
@jsl I think that should not be a problem since this is `postgresql-simple` wrapper so it's mean to be used only with PostgreSQL, on the other hand I like the idea of the `.tables` file since this query may not be compatible with all postgres versions